### PR TITLE
feat(data): clean pantip 2G dataset

### PR DESCRIPTION
**Pull request by boss-chanon**
## Why this PR
Why we need this PR?
- This is a code to clean pantip 2G datasets to jsonl

## Changes
- add pipeline for clean pantip 2G

## Related Issues
Close # 160

## Checklist
- [ ] PR should be in the [Naming convention](../../docs/PR_NAMING.md)
- [ ] Assign yourself in to Assigneees
- [ ] Tag related issues
- [ ] Constants name should be ALL_CAPITAL, function name should be snake_case, and class name should be CamelCase
- [ ] complex function/algorithm should have [Docstring](https://peps.python.org/pep-0257/)
- [ ] 1 PR should not have more than 200 lines changes (Exception for test files). If more than that please open multiple PRs
- [ ] At least PR reviewer must come from the task's team (model, eval, data)

### CODE REVIEW

File: src/data/scripts/pantip_2G/clean_data.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+parser.add_argument(
+    "--output_folder",
+    help='Name of an output folder (Default: "clean.txt")',
+    default="clean",
+)
+
+args = parser.parse_args()
+
+
+def clean_data(text):
```
Comment (boat1603) [inline src/data/scripts/pantip_2G/clean_data.py:22]: Could you write how this function do?

File: src/data/scripts/pantip_2G/clean_data.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    text = re.sub(r"\\\(.*?\^\-.*?\^\\\)", "", text)
+    text = emoji.demojize(text)
+    text = html.unescape(text)
+    text = re.sub(r"<.*?>", "", text)
+    text = text.strip()
+
+    return text
+
+
+def reformat_jsonl(input_file, output_file, source):
```
Comment (boat1603) [inline src/data/scripts/pantip_2G/clean_data.py:36]: This too

File: src/data/scripts/pantip_2G/clean_data.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        for item in reader.iter(skip_invalid=True):
+            tid = item["tid"]
+            cid = item["cid"]
+            desc = item.get("desc")
+
+            if tid != current_tid:
+                # Write the previous line (if any) to the output file
+                if current_tid is not None:
+                    data = {
+                        "source": source,
```
Comment (boat1603) [inline src/data/scripts/pantip_2G/clean_data.py:52]: Please use CONST

File: src/data/scripts/pantip_2G/clean_data.py (hunk 1 above)
Comment (boss-chanon) [reply src/data/scripts/pantip_2G/clean_data.py:28]: done

File: src/data/scripts/pantip_2G/clean_data.py (hunk 3 above)
Comment (boss-chanon) [reply src/data/scripts/pantip_2G/clean_data.py]: done

File: src/data/scripts/pantip_2G/clean_data.py (hunk 2 above)
Comment (boss-chanon) [reply src/data/scripts/pantip_2G/clean_data.py:47]: done

**Review by new5558 — APPROVED** [review]

**Pull request merged**